lib/extensions/core_extensions/active_record/relation.rb

```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+module Extensions::CoreExtensions::ActiveRecord::Relation
+  def select_for_count
```

Comment:
what's this for?

Reply:
Fix for `.count` to work with `baby_squeel` scopes (Arel nodes)

https://github.com/rails/rails/blob/4-2-stable/activerecord/lib/active_record/relation/calculations.rb#L386

May be fixed in Rails 5. https://github.com/rails/rails/commit/b220c9f9b4cc5b17af40b09046501f3469929a11

Reply:
OK. so you will remove this when upgrading to Rails 5?  Otherwise, it should stay together with baby squeel adapter.

Reply:
This issue is independent of the `cancancan-baby_squeel` adapter.

https://github.com/rails/rails/issues/15138
https://github.com/rails/rails/issues/24809

May have to submit PR to or fork `baby_squeel` gem for a proper fix.

Reply:
But coursemology's functionality does not depend on this fix right?  The issue is only there when `cancancan-baby_squeel` is introduced, is that correct?

Reply:
No. The issue surfaced when `cancancan-squeel` and indirectly `cancancan`'s monkeypatching of ActiveRecord was removed.

Here is where `squeel` did the monkeypatching https://github.com/activerecord-hackery/squeel/blob/master/lib/squeel/adapters/active_record/relation_extensions.rb#L86

The monkeypatching I did is a simplified version of that but it works as a temporary fix for now.

Reply:
Some queries that will error (remove monkeypatch and run these in rails console):

- `Course.first.levels.count`
- `Course.first.course_users.ordered_by_experience_points.count`
- `User.ordered_by_name.count`

Seems to only occur when there are calculated attributes.

Update: just confirmed that this is due to ActiveRecord issue and not `baby_squeel` or `cancancan-baby_squeel`: https://github.com/aha-app/calculated_attributes#known-issues

> In Rails 4.x, you cannot call count on a relation with calculated attributes. This is because of an ActiveRecord issue that does not permit Arel nodes in the count method.

Reply:
👍 OK, thanks for the explanation, I finally get it, so Coursemology still depends on this to work.

Reply:
Possible to add this as a comment on why it is there, and pending what fixes?

Reply:
done